eva_submission/eload_ingestion.py

```diff
@@ -0,0 +1,111 @@
+        self.check_variant_db(db_name)
+
+        if not tasks:
+            tasks = self.all_tasks
+
+        if 'metadata_load' in tasks:
+            self.load_from_ena()
+        if 'accession' in tasks:
+            self.warning('Accessioning not yet supported, skipping.')
+        if 'variant_load' in tasks:
+            self.warning('Variant loading not yet supported, skipping.')
+
+    def get_db_name(self):
+        """
+        Constructs the expected database name in mongo, based on assembly info retrieved from ENA.
+        """
+        assm_accession = self.eload_cfg.query('submission', 'assembly_accession')
+        ena_url = f'https://www.ebi.ac.uk/ena/browser/api/xml/{assm_accession}'
+        try:  # catches any kind of request error, including non-20X status code
+            response = requests.get(ena_url)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            self.error(f"Couldn't get assembly info from ENA for accession {assm_accession}")
+            raise e
+
+        soup = BeautifulSoup(response.text, 'lxml')
+        sci_name_terms = soup.scientific_name.text.split()
+        sci_name = sci_name_terms[0][0] + ''.join(sci_name_terms[1:])
+        assembly_name = soup.find('name').text
+        return f'eva_{sanitize(sci_name)}_{sanitize(assembly_name)}'
+
+    def check_variant_db(self, db_name=None):
+        """
+        Checks mongo for the right variant database.
+        If db_name is provided it will check for that, otherwise it will construct the expected database name.
+        """
+        #
+        if not db_name:
+            db_name = self.get_db_name()
+        self.eload_cfg.set(self.config_section, 'database', 'db_name', value=db_name)
+
+        with get_mongo_connection_handle(
+                username=cfg.query('mongo', 'username'),
+                password=cfg.query('mongo', 'password'),
+                host=cfg.query('mongo', 'host')
+        ) as db:
+            names = db.list_database_names()
+            if db_name in names:
+                self.info(f'Found database named {db_name}.')
+                self.info('If this is incorrect, please cancel and pass in the database name explicitly.')
+                self.eload_cfg.set(self.config_section, 'database', 'exists', value=True)
+            else:
+                self.error(f'Database named {db_name} does not exist in variant warehouse, aborting.')
+                self.error('Please create the database or pass in the appropriate database name explicitly.')
+                self.eload_cfg.set(self.config_section, 'database', 'exists', value=False)
+                raise ValueError(f'No database named {db_name} found.')
+
+    def load_from_ena(self):
+        """
+        Loads project metadata from ENA into EVADEV.
+        """
+        project_accession = self.eload_cfg.query('brokering', 'ena', 'PROJECT')
```

Comment:
I realise that I did not mention this during our discussion but there are cases where we only pull one `ANALYSIS` from ENA instead of the whole `PROJECT`. 
@BaronKoy what is the general use case for this script ? Should we always load the whole project or always specify the analysis (There might be several then).

Reply:
As discussed, we currently always use project rather than specifying analysis.